File: iut/TP5_TP6/impot.py


def montant_impot(R, n):

    res = R/n
    if (res >= 500):
        R = R * 0.2
    elif (res < 500):
        R = R * 0.1
    return R


def revenu_net(R, n):
    return R - montant_impot(R,n)
    


def impot(R, n):

    montant = montant_impot(R,n)
    impot = revenu_net(R,n)
    print(f"\nLe montant des impots est de : {montant}")
    print(f"Le revenue net est de : {impot}")


R = float(input("Revenue : "))
n = int(input("Nombres de membres du foyer : "))
impot(R,n)

File: iut/TP5_TP6/impot.c
#include <stdio.h>

double montant_impot(double R, double n)
{
    double res;

    res = R/n;
    if (res >= 500)
        R = R * 0.2;
    else if (res < 500)
        R = R * 0.1;
    return R;
}

double revenu_net(double R, double n)
{
    double res;

    res = R - montant_impot(R,n);
    return res;
}

void impot(double R, int n)
{
    double montant, impot;

    montant = montant_impot(R,n);
    impot = revenu_net(R,n);
    printf("\nLe montant des impots est de : %f\n", montant);
    printf("Le revenue net est de : %f", impot);
}

int main()
{
    int n;
    double R;
    printf("Revenue : ");
    scanf("%lf",&R);
    printf("Nombres de membres du foyer : ");
    scanf("%i", &n);
    impot(R,n);
    return 0;
}
